internal/sirius/my_details.go

```diff
@@ -45,9 +49,47 @@ func (c *Client) MyDetails(ctx context.Context, cookies []*http.Cookie) (MyDetai
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		return v, errors.New("returned non-2XX response")
+		return v, errors.New("returned non-2XX response: " + strconv.Itoa(resp.StatusCode))
 	}
 
 	err = json.NewDecoder(resp.Body).Decode(&v)
 	return v, err
 }
+
+func (c *Client) EditMyDetails(ctx context.Context, cookies []*http.Cookie, id int, phoneNumber string) (ValidationErrors, error) {
```

Comment:
Kind of weird having a method return 2 errors. You could make `ValidationErrors` satisfy the `error` interface by adding a `Error() string` method to it. Then if you want to get at the actual value do something like:

```go
err := EditMyDetails(blah...)
if err == ErrUnauthorized {
  // send to login
} else if verr, ok := err.(ValidationErrors); ok {
  // use verr in template
} else if err != nil {
  // some other error
}
```

Reply:
I've created `ValidationError`, which satisifies `error` and also has an `Errors` field containing the individual field-level errors. As suggested, the handler identifies it and shows responses a bit differently.

This seems much more appropriate, though the naming system of `sirius.ValidationError` (the error type), `sirius.ValidationErrors` (the map of field-level errors) and the `Error` field is a bit ugly and I'm quite happy to change it 😄